src/components/pages/tx/transaction-details.tsx

```diff
@@ -7,38 +8,58 @@ import { Card, CardContent } from "@/components/ui/card";
 import { Transaction } from "@/lib/types";
 import { formatGwei, formatEther } from "viem";
 
-const TransactionDetails = ({ transaction }: { transaction: Transaction }) => (
-  <Card>
-    <CardContent className="p-4">
-      <dl>
-        <DescriptionListItem title="Transaction Hash">
-          {transaction.hash}
-        </DescriptionListItem>
-        <DescriptionListItem title="Block">
-          <Link
-            className="text-primary hover:brightness-150"
-            href={`/block/${transaction.blockNumber}`}
-          >
-            {transaction.blockNumber.toString()}
-          </Link>
-        </DescriptionListItem>
-        <DescriptionListItem title="Timestamp">
-          <Clock className="mr-1 size-4" />
-          {formatTimestamp(transaction.timestamp)}
-        </DescriptionListItem>
-        <DescriptionListItem border title="Value">
-          <EthereumIcon className="mr-1 size-4" />
-          {formatEther(transaction.value)} ETH
-        </DescriptionListItem>
-        <DescriptionListItem title="Gas Price">
-          {formatGwei(transaction.gasPrice ?? BigInt(0))} Gwei
-          <span className="ml-1 text-muted-foreground">
-            ({formatEther(transaction.gasPrice ?? BigInt(0))} ETH)
-          </span>
-        </DescriptionListItem>
-      </dl>
-    </CardContent>
-  </Card>
-);
+const TransactionDetails = ({ transaction }: { transaction: Transaction }) => {
```

Comment:
Add a top-level isContractInteraction prop here on the TransactionDetails component that will be computed one level up in the root component: https://github.com/walnuthq/op-scan/blob/main/src/components/pages/tx/index.tsx#L6-L7

Compute the property and pass it to this component along with the transaction.
Replicate the logic you wrote before by calling getCode on the to address of the tx.